### PR TITLE
BKNDLSS-29165 Stripe Payment

### PR DESCRIPTION
**Pull request by serhmel**
Screenshot_1  Screenshot_2 
Screenshot_6 Screenshot_7 Screenshot_9
Screenshot_3 Screenshot_4 Screenshot_5

UI Preview:
ui_preview

### CODE REVIEW

File: components/bl-stripe-payment-component/src/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const [stripePromise, setStripePromise] = useState(() => {
+    return loadStripe(publishableKey);
+  });
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn(classList, 'bl-customComponent-stripe-payment') }>
```
Comment (Saharij) [inline components/bl-stripe-payment-component/src/index.js:26]: In my opinion, it is better when first there will be a main class ('bl-customComponent-stripe-payment'), and then a classList
`cn('bl-customComponent-stripe-payment', classList)`

File: components/bl-stripe-payment-component/src/payment-components/billing-details.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  if (!billingDetailsVisibility) {
+    return null;
+  }
+
+  return (
+    <div className="billing-details">
+      { FormFields.map((field, index) => (
+        <FormField
+          visibility={ fieldsVisibility[field.name] }
+          key={ index }
```
Comment (Saharij) [inline components/bl-stripe-payment-component/src/payment-components/billing-details.js:35]: The property `key` must be the first

File: components/bl-stripe-payment-component/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+<dt>Min Amount</dt>
+<dd>Value from this field will be used as the minimum amount that can be charged. Defaults to 0.5.</dd>
+<dt>Fixed Amount</dt>
+<dd>Checkbox for determining whether the payment amount is fixed or can be changed by the user. Checked by default.</dd>
+<dt>Currency</dt>
+<dd>Value from this field will be used as the currency label. Defaults to USD</dd>
+<dt>Name Visibility</dt>
+<dd>Checkbox for determining whether the payment form contains a field for the cardholder's name. Unchecked by default.</dd>
+<dt>Address Visibility</dt>
+<dd>Checkbox for determining whether the payment form contains a field for the address. Unchecked by default.</dd>
```
Comment (v-excelsior) [inline components/bl-stripe-payment-component/README.md:27]: Could we describe checkbox value using booleans? true/false

File: components/bl-stripe-payment-component/src/helpers/use-card-element.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    const formField = event.target;
+    const billingDetails = {
+      'address_city'   : formField.city ? formField.city.value : undefined,
```
Comment (v-excelsior) [inline components/bl-stripe-payment-component/src/helpers/use-card-element.js:26]:
```suggestion
      'address_city'   : formField?.city?.value,
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

And for others

File: components/bl-stripe-payment-component/src/payment-form.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      backgroundColor: 'transparent',
+      color          : cardElementColor,
+    },
+  };
+
+  const cardElementOptions = {
+    iconStyle     : 'solid',
+    style         : cardElementStyles,
+    hidePostalCode: true,
+  };
```
Comment (v-excelsior) [inline components/bl-stripe-payment-component/src/payment-form.js:38]: move into function below the component

File: components/bl-stripe-payment-component/src/payment-form.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        <CardElement
+          id="card-element"
+          options={ cardElementOptions }
+          onFocus={ onCardFocus }
+          onBlur={ onCardBlur }
+          onChange={ onCardChange }
+        />
+      </div>
+      <PaymentAmount component={ component }/>
+      <button type="submit" disabled={ isDisabled }>{ `Pay ${ amount || 0 } ${ currency || '' }` }</button>
```
Comment (v-excelsior) [inline components/bl-stripe-payment-component/src/payment-form.js:64]: `isDisabled` -> `disabled` or `whatDisabled`(e.g. `veryDangerousButtonDisabled`)

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-stripe-payment-component/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+## Usage
+
+Before adding the component to the page, open the Marketplace screen and install
+the [Stripe Integration Plugin](https://backendless.com/features/marketplace/stripe-integration-plugin/) from the
+Backendless marketplace. The plugin product is available under the All Services menu.
+
+Then add the component to the page and specify the Publishable key in UI-Builder Settings.
+
+### Styles and Settings
```
Comment (Valodya) [inline components/bl-stripe-payment-component/README.md:20]: please change the readme according to the new standard https://github.com/Backendless/ui-builder-library/blob/main/components/bl-leafletMap-component/README.md

File: components/bl-stripe-payment-component/component.json [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
...etween;\"><i data-icon=\"credit_card\" data-size=\"default\" data-module-type=\"system\" data-module-id=\"icon\" data-display data-uid=\"4c2bf12ca10afcd96a48dad57019e9b4\" style=\"margin:0 10px 4px 0;align-self:flex-end;\"></i><div data-label=\"\" data-variant=\"standard\" data-module-type=\"system\" data-module-id=\"input\" data-display data-uid=\"ed1f15d13f8b2736e855bd6bbbfdddc1\" data-value=\"\" data-placeholder=\"Card number\" data-readonly=\"false\" style=\"margin:0 0 0 0;width:100%;\"></div><div data-label=\"\" data-variant=\"standard\" data-module-type=\"system\" data-module-id=\"input\" data-display data-uid=\"369e4fde2c9c3b63ddfb4dd36d3a3174\" data-placeholder=\"MM/YY\" style=\"margin:0 0 0 0;\"></div><div data-label=\"\" data-variant=\"standard\" data-module-type=\"system\" data-module-id=\"input\" data-display data-uid=\"1e8b1b6e3d68ff0140dcc4bffdb24103\" data-placeholder=\"CVC\" style=\"margin:0 0 0 0;\"></div></div><button type=\"button\" data-label=\"Pay\" data-variant=\"contained\" data-size=\"medium\" data-module-type=\"system\" data-module-id=\"button\" data-display data-uid=\"6b9a68a23f21ca5734ed1f0ec55f24c9\" style=\"margin:10px 10px 10px 10px;\"></button></div>"
```
Comment (Valodya) [inline components/bl-stripe-payment-component/component.json:172]: I believe it should not be here

File: components/bl-stripe-payment-component/src/helpers/use-card-element.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    };
+
+    Backendless.CustomServices.invoke('Stripe', 'charge', chargeData)
+      .then(transaction => {
+        setTransactionDetails({ id: transaction.id, status: transaction.status });
+        cardElement.clear();
+        formRef.current.reset();
+        setIsLoading(false);
+
+        if (onSuccessEvent) {
```
Comment (Valodya) [inline components/bl-stripe-payment-component/src/helpers/use-card-element.js:57]: not sure, but I guess all defined event handlers always exist

**Pull request merged**